lib/MusicBrainz/Server/Controller/Relationship/LinkAttributeType.pm

```diff
@@ -77,24 +78,48 @@ sub edit : Chained('load') RequireAuth(relationship_editor)
 
     my $form = $c->form( form => 'Admin::LinkAttributeType', init_object => $link_attr_type );
 
+    my $attribute_in_use = $c->model('LinkAttributeType')->in_use($link_attr_type->id);
+
+    if ($attribute_in_use) {
+        $form->field('free_text')->disabled(1);
+        if ($link_attr_type->creditable) {
+            $form->field('creditable')->disabled(1);
+        }
+    }
+
     if ($c->form_posted_and_valid($form)) {
-        $c->model('MB')->with_transaction(sub {
-            $self->_insert_edit(
-                $c, $form,
-                edit_type => $EDIT_RELATIONSHIP_ATTRIBUTE,
-                entity_id => $link_attr_type->id,
-                new => { map { $_->name => $_->value } $form->edit_fields },
-                old => {
-                    name => $link_attr_type->name,
-                    description => $link_attr_type->description,
-                    parent_id => $link_attr_type->parent_id,
-                    child_order => $link_attr_type->child_order,
+        my %values = map { $_->name => $_->value } $form->edit_fields;
+
+        my $removes_creditable = $link_attr_type->creditable && !($values{'creditable'});
+        my $adds_free_text = $values{'free_text'} && !($link_attr_type->free_text);
+        my $removes_free_text = $link_attr_type->free_text && !($values{'free_text'});
+        if ($attribute_in_use && $removes_creditable) {
+            $form->field('creditable')->add_error(l('The attribute is in use, and this change would risk damaging existing values'));
+        } elsif ($attribute_in_use && ($adds_free_text || $removes_free_text)) {
+            $form->field('free_text')->add_error(l('The attribute is in use, and this change would risk damaging existing values'));
```

Comment:
If you disable the checkboxes, this shouldn't be possible, so we can probably remove the error messages. But I do think we should perform these checks in `Data::LinkAttributeType::update` and `die` if they occur.